### PR TITLE
feat: manually trigger periodic task

### LINKED ISSUE #415: Allow manual trigger of interval tasks
## Feature description 

Was running BDjuno but it hasn't been a day yet, I want to manually trigger the calculate inflation function but can't seem to be able to find a way to do it. Perhaps BDjuno can allow user trigger interval tasks manually, so they can be triggered on demand, I feel this would be especially useful during development.

Especially tasks that runs less frequently than hourly, like inflation and community pool:
https://github.com/forbole/bdjuno/blob/cosmos/v0.44.x/Requirements.md#on-intervals

## Implementation proposal

Add new commands to CLI so users can trigger interval tasks manually.

Would love to help with implementing this if you think this is worth implementing.

### PR DESCRIPTION
**Pull request by huichiaotsou**
## Description

Closes: #415 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch
- [x] provided a link to the relevant issue or specification
- [x] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by MonikaCat** [comment]
Hey @huichiaotsou LGTM but I think we can remove the periodic-operations cmd, move cmds inside it to separate modules and update those cmds to let users know which data is being updated so it would look like this eg. `bdjuno parse mint inflation` , `bdjuno parse distribution community-pool`

**Comment by huichiaotsou** [comment]
> Hey @huichiaotsou LGTM but I think we can remove the periodic-operations cmd, move cmds inside it to separate modules and update those cmds to let users know which data is being updated so it would look like this eg. `bdjuno parse mint inflation` , `bdjuno parse distribution community-pool`

@MonikaCat I've done the modifications :D

File: cmd/parse/mint/inflation.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	parsecmdtypes "github.com/forbole/juno/v3/cmd/parse/types"
+	"github.com/forbole/juno/v3/types/config"
+	"github.com/spf13/cobra"
+
+	"github.com/forbole/bdjuno/v3/database"
+	"github.com/forbole/bdjuno/v3/modules/mint"
+	modulestypes "github.com/forbole/bdjuno/v3/modules/types"
+)
+
+// inflationCmd returns the Cobra command allowing to refresh x/mint infaltion
```
Comment (MonikaCat) [inline cmd/parse/mint/inflation.go]:
```suggestion
// inflationCmd returns the Cobra command allowing to refresh x/mint inflation
```

File: cmd/parse/mint/inflation.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+func inflationCmd(parseConfig *parsecmdtypes.Config) *cobra.Command {
+	return &cobra.Command{
+		Use:   "inflation",
+		Short: "Refresh infaltion",
```
Comment (MonikaCat) [inline cmd/parse/mint/inflation.go]:
```suggestion
		Short: "Refresh inflation",
```

File: cmd/parse/staking/staking.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+			// Get the database
+			db := database.Cast(parseCtx.Database)
+
+			// Build staking module
+			stakingModule := staking.NewModule(sources.StakingSource, parseCtx.EncodingConfig.Marshaler, db)
+
+			err = stakingModule.UpdateStakingPool()
+			if err != nil {
+				return err
```
Comment (MonikaCat) [inline cmd/parse/staking/staking.go]:
```suggestion
				return fmt.Errorf("error while updating staking pool: %s", err)
```

File: cmd/parse/mint/inflation.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+			// Get the database
+			db := database.Cast(parseCtx.Database)
+
+			// Build mint module
+			mintModule := mint.NewModule(sources.MintSource, parseCtx.EncodingConfig.Marshaler, db)
+
+			err = mintModule.UpdateInflation()
+			if err != nil {
+				return err
```
Comment (MonikaCat) [inline cmd/parse/mint/inflation.go]:
```suggestion
				return fmt.Errorf("error while updating inflation: %s", err)
```

File: cmd/parse/distribution/communitypool.go [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+			// Get the database
+			db := database.Cast(parseCtx.Database)
+
+			// Build distribution module
+			distrModule := distribution.NewModule(sources.DistrSource, parseCtx.EncodingConfig.Marshaler, db)
+
+			err = distrModule.GetLatestCommunityPool()
+			if err != nil {
+				return err
```
Comment (MonikaCat) [inline cmd/parse/distribution/communitypool.go]:
```suggestion
				return fmt.Errorf("error while updating community pool: %s", err)
```

File: modules/bank/handle_periodic_operations.go [hunk 6]
```diff
@@ -23,19 +23,19 @@ func (m *Module) RegisterPeriodicOperations(scheduler *gocron.Scheduler) error {
 }
 
 // updateSupply updates the supply of all the tokens
```
Comment (MonikaCat) [inline modules/bank/handle_periodic_operations.go]:
```suggestion
// UpdateSupply updates the supply of all the tokens
```

File: modules/pricefeed/handle_periodic_operations.go [hunk 7]
```diff
@@ -56,8 +56,8 @@ func (m *Module) getTokenPrices() ([]types.TokenPrice, error) {
 	return prices, nil
 }
 
-// updatePrice fetch total amount of coins in the system from RPC and store it into database
-func (m *Module) updatePrice() error {
+// UpdatePrice fetch total amount of coins in the system from RPC and store it into database
```
Comment (MonikaCat) [inline modules/pricefeed/handle_periodic_operations.go]:
```suggestion
// UpdatePrice fetches the total amount of coins in the system from RPC and stores it in database
```

**Review by MonikaCat — APPROVED** [review]
tested, LGTM

**Review by rmontagnin — APPROVED** [review]
utACK

**Pull request merged**